polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1087,7 +1095,10 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				tracing::trace!(target: "xcm::executor::BuyExecution", asset_used_for_fees = ?self.asset_used_for_fees);
 				// pay for `weight` using up to `fees` of the holding register.
 				let max_fee =
-					self.holding.try_take(fees.into()).map_err(|_| XcmError::NotHoldingFees)?;
+					self.holding.try_take(fees.clone().into()).map_err(|e| {
+						log::error!(target: "xcm::process_instruction::buy_execution", "Failed to take fees {fees:?} from holding with error: {e:?}");
```

Comment:
@ayevbeosa can you please:
- [ ] change all `log::error` to `tracing::error` format for `xcm-executor`
- [ ] check existing logs, e.g. this one can be extended with more info `tracing::error!(target: "xcm::reanchor", ?error, "Failed reanchoring with error");`

Reply:
I will get on it